lib/main.js

```diff
@@ -54,7 +54,9 @@ module.exports = {
 
     this.subscriptions.add(atom.workspace.observeTextEditors(editor => {
       editor.onDidSave(() => {
-        if (scopes.indexOf(editor.getGrammar().scopeName) !== -1 && atom.config.get('linter-eslint.fixOnSave')) {
+        const editorScopes = editor.getLastCursor().getScopeDescriptor().scopes;
+        const validScope = editorScopes.some(scope => scopes.includes(scope));
```

Comment:
Actually since currently this code is using an `atom.config.get()` call this is _much_ cheaper than checking that setting value since that has to do a ton of resolving and checking of things before it returns a value. When that gets moved to an `observe` in a future code overhaul then it will be better to check it first.